### PR TITLE
driver error handling

### PR DESCRIPTION
**Pull request by vkagamlyk**

### CODE REVIEW

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java [hunk 1]
```diff
@@ -69,6 +71,20 @@ public class HttpHandlerUtil {
...
+    /**
+     * Helper method to send errors back as JSON. Only to be used when the RequestMessage couldn't be parsed, because
+     * a proper serialized ResponseMessage should be sent in that case.
+     *
+     * @param ctx           The netty channel context.
+     * @param status        The HTTP error status code.
+     * @param message       The error message to contain the body.
+     * @param closeChannel  Close the netty channel after error message.
+     */
+    public static void sendError(final ChannelHandlerContext ctx, final HttpResponseStatus status, final String message,
```
Comment (kenhuuu) [inline gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java:86]: When is closing the channel explicitly needed?

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java [hunk 2]
```diff
@@ -112,14 +115,17 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
...
-                sendError(ctx, msg);
+                sendError(ctx, msg, ae);
             }
         }
     }
 
-    private void sendError(final ChannelHandlerContext ctx, final Object msg) {
-        // Close the connection as soon as the error message is sent.
-        ctx.writeAndFlush(new DefaultFullHttpResponse(HTTP_1_1, UNAUTHORIZED)).addListener(ChannelFutureListener.CLOSE);
+    private void sendError(final ChannelHandlerContext ctx, final Object msg, final Throwable t) {
```
Comment (kenhuuu) [inline gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java:123]: Could we reuse HttpHandlerUtil's sendError here?

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java [hunk 3]
```diff
@@ -64,32 +70,29 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
         if (msg instanceof FullHttpMessage) {
             final FullHttpMessage request = (FullHttpMessage) msg;
             if (!request.headers().contains("Authorization")) {
-                sendError(ctx, msg);
+                sendError(ctx, msg, "Missing or incorrect credentials");
```
Comment (kenhuuu) [inline gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java:73]: Nit: make static String for "Missing or...".

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java (hunk 1 above)
Comment (vkagamlyk) [reply gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java:86]: removed.

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java (hunk 2 above)
Comment (vkagamlyk) [reply gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java:123]: reused

**Comment by kenhuuu** [comment]
LGTM

**Pull request merged**